src/types/ast/rational_literal.ts

```diff
@@ -0,0 +1,29 @@
+import { Range } from "../../misc";
+import { TypeNode } from "./type";
+
+interface Rational {
+  numerator: bigint;
+  denominator: bigint;
+}
+
+export class RationalLiteralType extends TypeNode {
+    public readonly literal?: Rational;
```

Comment:
Question: Why is the literal member optional? Looking at the changes in  the grammar in typeString_grammar.pegjs literal is always provided.

Reply:
I had this same question about `IntLiteralType` and presumed there was some usecase I wasn't aware of.

Reply:
Addressed in https://github.com/ConsenSys/solc-typed-ast/pull/98/commits/19f312848b466885b4cdcb3fff2eb5dd7ec6c813